data/src/test/java/io/homeassistant/companion/android/data/integration/IntegrationRepositoryImplSpec.kt

```diff
@@ -105,7 +112,300 @@ object IntegrationRepositoryImplSpec : Spek({
+                    4,
+                    5
+                )
+                val integrationRequest = IntegrationRequest(
+                    "update_location",
+                    UpdateLocationRequest(
+                        location.locationName,
+                        location.gps,
+                        location.gpsAccuracy,
+                        location.battery,
+                        location.speed,
+                        location.altitude,
+                        location.course,
+                        location.verticalAccuracy
+                    )
+                )
+                beforeEachTest {
+                    coEvery { localStorage.getString("cloud_url") } returns null
+                    coEvery { localStorage.getString("remote_ui_url") } returns null
+                    coEvery {
+                        integrationService.updateLocation(
+                            any(), // "http://example.com/api/webhook/FGHIJ",
+                            any() // integrationRequest
+                        )
+                    } returns Response.success(null)
+                    runBlocking { repository.updateLocation(location) }
+                }
+
+                it("should call the service.") {
+                    coVerify {
+                        integrationService.updateLocation(
+                            "http://example.com/api/webhook/FGHIJ".toHttpUrl(),
+                            integrationRequest
+                        )
+                    }
+                }
+            }
+
+            describe("updateLocation fail then succeeds") {
+                val location = UpdateLocation(
+                    "locationName",
+                    arrayOf(45.0, -45.0),
+                    0,
+                    1,
+                    2,
+                    3,
+                    4,
+                    5
+                )
+                val integrationRequest = IntegrationRequest(
+                    "update_location",
+                    UpdateLocationRequest(
+                        location.locationName,
+                        location.gps,
+                        location.gpsAccuracy,
+                        location.battery,
+                        location.speed,
+                        location.altitude,
+                        location.course,
+                        location.verticalAccuracy
+                    )
+                )
+
+                beforeEachTest {
+                    coEvery { localStorage.getString("cloud_url") } returns "http://best.com/hook/id"
+                    coEvery { localStorage.getString("remote_ui_url") } returns "http://better.com"
+                    coEvery {
+                        integrationService.updateLocation(
+                            "http://best.com/hook/id".toHttpUrl(),
+                            any() // integrationRequest
+                        )
+                    } throws Exception()
```

Comment:
Ok so here is my point :
If you replace the `throws Exception()` by 
````
                    } returns mockk {
                        every { isSuccessful } returns false
                    }
````
The test wont pass. 
But I think we should have the same behaviour ? (if it's the case could you add this test and fix the code)